src/Widgets/AppButton.vala

```diff
@@ -147,6 +147,22 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
             return false;
         });
 
+        this.key_press_event.connect ((e) => {
+            if (e.keyval == Gdk.Key.Menu) {
+                menu = new Slingshot.AppContextMenu (desktop_id, desktop_path);
+                menu.app_launched.connect (() => {
+                    app_launched ();
+                });
+
+                if (menu != null && menu.get_children () != null) {
```

Comment:
I suppose `menu` after creation will never be null, so you could remove this check here (as well as in the original code).